policyengine_us/variables/gov/states/mn/tax/income/niit/mn_niit.py

```diff
@@ -0,0 +1,47 @@
+from policyengine_us.model_api import *
+
+class mn_niit(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Minnesota Net Investment Income Tax (NIIT)"
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "https://www.revenue.state.mn.us/sites/default/files/2024-12/m1-24.pdf"
+        "https://www.revisor.mn.gov/statutes/cite/290.033"
+    )
+    defined_for = StateCode.MN
+
+    def formula(tax_unit, period, parameters):
+        """
+        Computes Minnesota's net investment income tax using a bracket structure.
+        Currently, the brackets are defined such that:
+          - Amounts up to $1,000,000 are taxed at 0%
+          - Amounts above $1,000,000 are taxed at 1%
+        """
+        p = parameters(period).gov.states.mn.tax.income.niit
+        net_investment_income = tax_unit("net_investment_income", period)
+        total_niit = 0.0
+        previous_threshold = 0
+        for i, bracket in enumerate(p):
```

Comment:
can we use scale parameters here? https://openfisca.org/doc/coding-the-legislation/legislation_parameters.html#creating-scales